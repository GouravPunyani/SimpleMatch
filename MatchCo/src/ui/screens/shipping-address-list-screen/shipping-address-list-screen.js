// @flow

import React, { Component } from 'react'
import { View, Text, StyleSheet, Dimensions } from 'react-native'
import { inject, observer } from 'mobx-react'
import { Button, ButtonPanel, StaticLayout } from '../..'
import type { UserStore } from '../../../stores/user-store'
import { translator } from '../../../services/translate/translator'
const { width, height } = Dimensions.get('window')

@translator
@inject('userStore')
@observer
export class ShippingAddressListScreen extends Component {
  static propTypes: {
    navigation: any
  };

  render () {
    const { userStore, navigation, translate } = this.props
    const navigate = (path, options = {}) => () => navigation.navigate(path, options)

    return (
      // TODO: Allow for multiple shipping addresses
      <StaticLayout>
        <ButtonPanel>
          <Text style={styles.title}>{translate('account.shippingAddress')}</Text>
          <View style={{width: width*.9, height: 1, backgroundColor: '#dcb8bd'}}/>
          <View style={styles.wrapper}>
            <View>
              <View style={{flexDirection:'row'}}>
                <Text style={styles.text}>{'Matthieu'/* <-- Will need to be taken from account*/}</Text>
                <Text style={styles.text}>{'Rey'/* <-- Will need to be taken from account*/}</Text>
              </View>
              <Text style={styles.text}>{'1421 5th Street'/* <-- Will need to be taken from account*/}</Text>
              <View style={{flexDirection:'row'}}>
                <Text style={styles.text}>{'Santa Monica'/* <-- Will need to be taken from account*/}</Text>
                <Text style={styles.text}>{'CA'/* <-- Will need to be taken from account*/}</Text>
                <Text style={styles.text}>{'90401'/* <-- Will need to be taken from account*/}</Text>
              </View>
              <Text style={styles.text}>{'123.456.7890'/* <-- Will need to be taken from account*/}</Text>
            </View>
            <Button text={translate('account.update')} onPress={navigate('shippingAddress')}
            textStyle={{color: '#dcb8bd', textDecorationLine: 'underline', fontSize: 12}}
            style={{minHeight: 20, backgroundColor: 'transparent', shadowOpacity: 0}} />
          </View>
        </ButtonPanel>
      </StaticLayout>
    )
  }
}

const styles = StyleSheet.create({
  wrapper: { flexDirection: 'row', justifyContent: 'space-between', paddingVertical: 7, paddingHorizontal: 5 },
  button: {minHeight: 20, backgroundColor: 'transparent', shadowOpacity: 0},
  buttonText: {color: 'dimgray', textDecorationLine: 'underline', fontSize: 12},
  title: { color: 'black', fontSize: 14, padding: 7 },
  text: { color: 'dimgray', fontSize: 12, paddingHorizontal: 2 }
})
