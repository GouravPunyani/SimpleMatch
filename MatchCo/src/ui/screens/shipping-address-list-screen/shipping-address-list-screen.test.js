import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { ShippingAddressListScreen as Screen } from './shipping-address-list-screen'

test('renders correctly', () => {
  const tree = renderer.create(<Screen />)
  expect(tree).toMatchSnapshot()
})
