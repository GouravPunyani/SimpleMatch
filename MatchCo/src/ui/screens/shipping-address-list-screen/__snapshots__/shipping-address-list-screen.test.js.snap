// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rebeccapurple",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 30,
      }
    }
  >
    ShippingAddressListScreen
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 12,
      }
    }
  >
    src/ui/screens/shipping-address-list-screen/shipping-address-list-screen.js
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "margin": 10,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "margin": 4,
          "minHeight": 44,
          "minWidth": 44,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 4,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "Nexa",
              "fontSize": 14,
            },
          ]
        }
      >
        Shipping Address 1
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "margin": 4,
          "minHeight": 44,
          "minWidth": 44,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 4,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "Nexa",
              "fontSize": 14,
            },
          ]
        }
      >
        Shipping Address 2
      </Text>
    </View>
  </View>
</View>
`;
